### PR TITLE
Update recipient_create documentation

### PR DESCRIPTION
**Pull request by nkomanski**
# Description
- Update recipient_create documentation

# Steps to fix the test pipeline
- Added update_type property for Item in models.rb
- Remove test_credit_details.rb file. This endpoint doesn't exist any more.
- Remove test_income.rb file. This endpoint doesn't exist any more.
- Remove tests for create_payment_token_response since they were breaking the test pipeline.
- Change the start_date of standing orders tests to '9999-01-01'. This allows for live tests to work and prevents having to re-record tests with every new PR.
- Fix test_processor.rb
- Update VCR cassettes

### CODE REVIEW

**Comment by nkomanski** [comment]
I don't understand how this PR could make the test pipeline fail since it's just updating docs.

**Review by cgfarmer4 — APPROVED** [review]

**Review by skansal-plaid — APPROVED** [review]
The tests are failing because the VCR cassette expects schedule start date [hardcoded to `2021-01-14`](https://github.com/plaid/plaid-ruby/blob/39738297f95797774386c3dc3b48acf108b791b0/test/vcr_cassettes/PlaidPaymentInitiationTest_test_all_routes.yml#L8457). However, we are passing in a [variable date 7 days from now](https://github.com/plaid/plaid-ruby/blob/39738297f95797774386c3dc3b48acf108b791b0/test/test_payment_initiation.rb#L92). We should probably just fix [start date in the test file](https://github.com/plaid/plaid-ruby/blob/39738297f95797774386c3dc3b48acf108b791b0/test/test_payment_initiation.rb#L92) to `2021-01-14`.

Details:
```
An HTTP request has been made that VCR does not know how to handle:
  POST https://sandbox.plaid.com/payment_initiation/payment/create
  Body: {"recipient_id":"recipient-id-sandbox-e587d96c-f15b-4d42-b10f-5376065ce563","reference":"testpayment","amount":{"currency":"GBP","value":100.0},"schedule":{"interval":"MONTHLY","interval_execution_day":1,"start_date":"2021-02-16"},"client_id":"PLAID_RUBY_CLIENT_ID","secret":"PLAID_RUBY_SECRET"}
```

https://app.circleci.com/pipelines/github/plaid/plaid-ruby/205/workflows/8d629dae-8418-42a4-bf76-914a30cfb39d/jobs/667

**Review by jasonjia93 — CHANGES_REQUESTED** [review]
> The tests are failing because the VCR cassette expects schedule start date [hardcoded to `2021-01-14`](https://github.com/plaid/plaid-ruby/blob/39738297f95797774386c3dc3b48acf108b791b0/test/vcr_cassettes/PlaidPaymentInitiationTest_test_all_routes.yml#L8457). However, we are passing in a [variable date 7 days from now](https://github.com/plaid/plaid-ruby/blob/39738297f95797774386c3dc3b48acf108b791b0/test/test_payment_initiation.rb#L92). We should probably just fix [start date in the test file](https://github.com/plaid/plaid-ruby/blob/39738297f95797774386c3dc3b48acf108b791b0/test/test_payment_initiation.rb#L92) to `2021-01-14`.
> 
> Details:
> 
> ```
> An HTTP request has been made that VCR does not know how to handle:
>   POST https://sandbox.plaid.com/payment_initiation/payment/create
>   Body: {"recipient_id":"recipient-id-sandbox-e587d96c-f15b-4d42-b10f-5376065ce563","reference":"testpayment","amount":{"currency":"GBP","value":100.0},"schedule":{"interval":"MONTHLY","interval_execution_day":1,"start_date":"2021-02-16"},"client_id":"PLAID_RUBY_CLIENT_ID","secret":"PLAID_RUBY_SECRET"}
> ```
> 
> https://app.circleci.com/pipelines/github/plaid/plaid-ruby/205/workflows/8d629dae-8418-42a4-bf76-914a30cfb39d/jobs/667

That isn't what we want to do either. Hard-coding the start date to `2021-01-14` is going to cause the standing orders tests to all fail since we can't create payments in the past. We should keep the start date as is and follow the instructions here to update the VCR stubs: https://github.com/plaid/plaid-ruby/blob/master/CONTRIBUTING.md

**Comment by jasonjia93** [comment]
Also based on the comment here: https://github.com/plaid/plaid-go/pull/158/files#r577098062, do we need this PR?

**Comment by nkomanski** [comment]
> Also based on the comment here: https://github.com/plaid/plaid-go/pull/158/files#r577098062, do we need this PR?

I'm ok with removing the examples in the README file, but I think it still makes sense to update the docs in lib/plaid/products/payment_initiation.rb.

**Review by jasonjia93 — DISMISSED** [review]
🙌  pending tests and comment for the start date

**Comment by skansal-plaid** [comment]
I believe tests are failing now because a new update_type field was added to the Item object but the ruby clib hasn't been updated to use that yet.

The new ruby clib that uses OpenAPI removes the need for a payment initiation lib https://github.com/plaid/plaid-ruby/pull/324/files so we can skip this PR if you want.

**Review by jasonjia93 — APPROVED** [review]

File: test/test_payment_initiation.rb [hunk 1]
```diff
@@ -61,22 +61,6 @@ def test_all_routes
...
     refute_empty(create_link_token_response.expiration)
 
-    # create legacy payment_token (deprecated)
-    create_payment_token_response =
-      client.payment_initiation.create_payment_token(payment_id)
-    refute_empty(create_payment_token_response.payment_token)
-    refute_empty(create_payment_token_response.payment_token_expiration_time)
-
-    # get payment
-    get_payment_response = client.payment_initiation.get_payment(payment_id)
```
Comment (jasonjia93) [inline test/test_payment_initiation.rb:71]: I think all of this logic should still work I think?

Comment (nkomanski) [reply test/test_payment_initiation.rb:71]: Sorry, this is not a final version of this PR. I removed the tests that were failing to make sure I could get to a non-breaking pipeline. I'll fix this now. In the meantime, I'm setting the PR as draft.

File: test/test_credit_details.rb [hunk 2]
```diff
@@ -7,16 +7,9 @@ def setup
...
-    refute_empty(response.credit_details)
-
-    account_id = response.accounts[3].account_id
-    response = client.credit_details.get(
-      access_token,
-      account_ids: [account_id]
-    )
-    assert_equal(1, response.credit_details.size)
+    assert_raises(Plaid::InvalidRequestError) do
+      client.credit_details.get(access_token)
```
Comment (skansal-plaid) [inline test/test_credit_details.rb:11]: Why should this error out?

**Review by skansal-plaid — DISMISSED** [review]

File: test/test_credit_details.rb (hunk 2 above)
Comment (cgfarmer4) [reply test/test_credit_details.rb:11]: I would rm the test completely. These were just finally turned off. They've been deprecated for a while.

**Review by cgfarmer4 — DISMISSED** [review]

File: test/test_credit_details.rb (hunk 2 above)
Comment (nkomanski) [reply test/test_credit_details.rb]: I agree with removing them

**Review by skansal-plaid — APPROVED** [review]

**Pull request merged**